src/main/java/net/dries007/tfc/api/events/ProspectEvent.java

```diff
@@ -0,0 +1,56 @@
+package net.dries007.tfc.api.events;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.eventhandler.Event;
+
+/**
+ * Event fired when the Prospector's Pickaxe is used.
+ * Carries all the data relative to the result displayed to the player.
+ * One of the two subclasses will be used according to the logical side.
+ */
+public abstract class ProspectEvent extends Event {
+
+    public static class Server extends ProspectEvent {
+        public Server(EntityPlayer player, BlockPos pos, ResultType type, String ore, double score){
+            super(player, pos, type, ore, score);
+        }
+    }
+
+    public static class Client extends ProspectEvent {
+        public Client(EntityPlayer player, BlockPos pos, ResultType type, String ore, double score){
+            super(player, pos, type, ore, score);
+        }
+    }
+
+    public enum ResultType {
```

Comment:
Well yeah, but actually no. ProspectResult is not exposed as API, I cannot refer to that from the API package unless it gets moved to the API package as well.

Reply:
>  I cannot refer to that from the API package unless it gets moved to the API package as well.

As I explained in another comment: We have no separate API artifact, the API package is *not* independent from the rest of the code base, it's not a separate source set, and we have no plans to make it so. You can say this all you want, it doesn't make it true.

Reply:
Well, if it is not independent, and therefore you cannot abstract from the rest of the codebase, then it is not an _**Abstract** Programming Interface_, right now calling anything else, say, `foo` is more appropriate, at least `foo` isn't a deceiving name. Don't interpret me wrong here, I don't mean to sound like an arrogant ass, but this _does_ bother me.

So right now, like Tinker's Construct has (or at least used to have), what you have here is not an API but a library to interact with the mod. I'd suggest either fixing it to be what the name says it is, or renaming it to what it actually is (lib).

This being said, I have a couple questions:
- Is the existence of this class in the event a major blocker?
- Is this not-actually-an-API decision something that's set in stone? or do you guys simply not care for it? why was this path chosen? As someone who has to code against the entire codebase I'd rather not have to, being able to code against the API alone and then run against the entire thing would ease some pains and make everything more flexible on this side. I would guess others like me would agree.

Please note that I'm not trying to question your decisions, or force you to reconsider them, but if I am to line up with TFC's expectations I need to understand how things are done, and most importantly, **why.**

Thank you for your time :)

Reply:
Your botherment is duly noted.

To answer your questions,
- Yes, it makes more sense to be in `ItemProPick`.
- Yes, since multiple years ago I can't find the exact date.
- Nope.
- Because the effort to maintain an *actual* separate source set and artifact wasn't deemed worth it given the effort required, and the fact that the majority of dependencies would be addons which tend to want to compile against the whole mod for other reasons.

> I would guess others like me would agree.

Citation needed - in a year of addon development this is the first complaint I'm hearing about the matter.

Reply:
Very well, this is the way.